### DISCUSSION TITLE
Migrating from xunit.console to dotnet test

### DISCUSSION THREAD

**Discussion by dennisdoomen** [xunit/xunit#2523] [Question] [answered]
We're migrating an existing set of tests that heavily rely on `async` code on .NET Framework 4.8 from xUnit Console 2.4 to `dotnet test`, but are experiencing weird behavior where some tasks take a long time to complete and causing tests to fail. I've played around with [various settings](https://xunit.net/docs/configuration-files) to find a way to workaround that.

So now I'm wondering if there is any kind of behavioral changes between running xUnit tests through `dotnet test` versus using the xUnit console that could affect how `Tasks` complete?

**Comment by bradwilson** [comment, marked as answer]
I'm not aware of any situation where Tasks would take longer just because they're run in .NET Core instead of .NET Framework.

That said, the execution pipeline with `dotnet test` (aka VSTest) is significantly different than when we run our tests directly (for example, using `xunit.console`). It's hard to know whether those things are contributing to your specific issues without a repro project.
